packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,485 @@
+            variant: 'grid',
+          });
+          await flushPromises();
+
+          const recommendationContainer = element.shadowRoot.querySelector(
+            selectors.gridContainer
+          );
+          const recommendationItems = element.shadowRoot.querySelectorAll(
+            selectors.recommendationListResult
+          );
+
+          expect(recommendationContainer).not.toBeNull();
+          expect(recommendationItems.length).toBe(recommendationsArray.length);
+          recommendationItems.forEach((item, index) => {
+            expect(item.result.title).toEqual(
+              recommendationsArray[index].title
+            );
+            expect(item.result.clickUri).toEqual(
+              recommendationsArray[index].clickUri
+            );
+            expect(item.result.excerpt).toEqual(
+              recommendationsArray[index].excerpt
+            );
+            expect(item.engineId).toBe(defaultOptions.engineId);
+          });
+        });
+
+        describe('with a custom #recommendationsPerRow option', () => {
+          const customRecommendationsPerRow = 2;
+          it('should set the width of the recommendation items based on the number of #recommendationsPerRow value', async () => {
+            const expectedRecommendationWidth = `${100 / customRecommendationsPerRow}%`;
+            const element = createTestComponent({
+              ...defaultOptions,
+              variant: 'grid',
+              recommendationsPerRow: 2,
+            });
+            await flushPromises();
+
+            expect(element.style._values['--recommendationItemWidth']).toEqual(
+              expectedRecommendationWidth
+            );
+          });
+        });
+      });
+
+      describe('when the variant value is #carousel', () => {
+        it('should properly display the recommendations as a carousel', async () => {
+          const element = createTestComponent({
+            ...defaultOptions,
+            variant: 'carousel',
+          });
+          await flushPromises();
+
+          const recommendationContainer = element.shadowRoot.querySelector(
+            selectors.carouselContainer
+          );
+
+          expect(recommendationContainer).not.toBeNull();
+          expect(recommendationContainer.label).toBe(defaultOptions.label);
+          expect(recommendationContainer.itemsPerPage).toBe(
+            defaultOptions.recommendationsPerRow
+          );
+
+          const recommendationItems = element.shadowRoot.querySelectorAll(
+            selectors.recommendationListResult
+          );
+
+          recommendationItems.forEach((item, index) => {
+            expect(item.result.title).toEqual(
+              recommendationsArray[index].title
+            );
+            expect(item.result.clickUri).toEqual(
+              recommendationsArray[index].clickUri
+            );
+            expect(item.result.excerpt).toEqual(
+              recommendationsArray[index].excerpt
+            );
```

Comment:
Same comment here, we can directly compare the object result.

Reply:
not really as when comparing object with deep equality Jest gives you they keys where there is a difference between the two objects you are comparing.

Additionally, scalability  wise testing the whole object with deep equality is better. because for example if we find that there are X more keys we wanna  check, the test will not change, however if we keep testing key by key we would have X more lines to write.